int32 position
float32 speed
bool active